crates/nu-json/src/util.rs

```diff
@@ -207,18 +207,19 @@ impl<Iter: Iterator<Item = u8>> ParseNumber<Iter> {
                             }
                         }
 
-                        if is_float {
-                            Ok(Number::F64(
-                                res.parse::<f64>().expect("Internal error: json parsing"),
-                            ))
-                        } else if res.starts_with('-') {
-                            Ok(Number::I64(
-                                res.parse::<i64>().expect("Internal error: json parsing"),
-                            ))
-                        } else {
-                            Ok(Number::U64(
-                                res.parse::<u64>().expect("Internal error: json parsing"),
-                            ))
+                        if !is_float {
+                            if let Ok(n) = res.parse::<i64>() {
+                                if res.starts_with('-') {
```

Comment:
do you want to check for the `-` first then try to parse as i64, otherwise parse as u64?

Reply:
Works for me, thanks!